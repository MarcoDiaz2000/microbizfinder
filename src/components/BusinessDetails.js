import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { fetchBusinessDetails } from '../redux/business/businessDetails';

const BusinessDetails = () => {
  const { id } = useParams();
  const dispatch = useDispatch();

  const { isFetching, details, error } = useSelector((state) => state.businessDetails);

  useEffect(() => {
    dispatch(fetchBusinessDetails(id));
  }, [dispatch, id]);

  if (isFetching) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        Error:
        {error.message}
      </div>
    );
  }

  if (!details) {
    return <div>No details available</div>;
  }

  return (
    <div className="detail-container">
      <img className="detail-image" src={details.image_url} alt={details.name} />
      <div className="detail-header">
        <h1>{details.name}</h1>
      </div>
      <div className="detail-body">
        <p className="detail-item">
          <strong>Phone:</strong>
          {' '}
          {details.phone}
        </p>
        <p className="detail-item">
          <strong>Address:</strong>
          {' '}
          {details.location.address1}
          ,
          {' '}
          {details.location.city}
          ,
          {' '}
          {details.location.zip_code}
        </p>
        <p className="detail-item">
          <strong>Rating:</strong>
          {' '}
          {details.rating}
        </p>
        <p className="detail-item">
          <strong>Reviews:</strong>
          {' '}
          {details.review_count}
        </p>
      </div>
    </div>
  );
};

export default BusinessDetails;
