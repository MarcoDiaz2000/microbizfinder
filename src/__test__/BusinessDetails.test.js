import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import BusinessDetails from '../components/BusinessDetails';

const mockStore = configureStore([thunk]);

test('BusinessDetails matches snapshot', () => {
  const store = mockStore({
    businessDetails: {
      isFetching: false,
      details: {
        id: '1',
        name: 'Business 1',
        image_url: 'example.com/business1.jpg',
        rating: 4.5,
        review_count: 10,
        phone: '123-456-7890',
        location: {
          address1: '123 Main St',
          city: 'City',
          zip_code: '12345',
        },
      },
      error: null,
    },
  });

  const { container } = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/details/1']}>
        <Routes>
          <Route path="/details/:id" element={<BusinessDetails />} />
        </Routes>
      </MemoryRouter>
    </Provider>,
  );

  expect(container).toMatchSnapshot();
});
