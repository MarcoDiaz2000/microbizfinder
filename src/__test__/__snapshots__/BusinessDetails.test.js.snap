// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessDetails matches snapshot 1`] = `
<div>
  <div
    class="detail-container"
  >
    <img
      alt="Business 1"
      class="detail-image"
      src="example.com/business1.jpg"
    />
    <div
      class="detail-header"
    >
      <h1>
        Business 1
      </h1>
    </div>
    <div
      class="detail-body"
    >
      <p
        class="detail-item"
      >
        <strong>
          Phone:
        </strong>
         
        123-456-7890
      </p>
      <p
        class="detail-item"
      >
        <strong>
          Address:
        </strong>
         
        123 Main St
        ,
         
        City
        ,
         
        12345
      </p>
      <p
        class="detail-item"
      >
        <strong>
          Rating:
        </strong>
         
        4.5
      </p>
      <p
        class="detail-item"
      >
        <strong>
          Reviews:
        </strong>
         
        10
      </p>
    </div>
  </div>
</div>
`;
